code/modules/mob/living/carbon/human/species_types/dwarves.dm

```diff
@@ -0,0 +1,217 @@
+			message += pick(" By Armok!")
+	speech_args[SPEECH_MESSAGE] = trim(message)
+
+//This mostly exists because my testdwarf's liver died while trying to also not die due to no alcohol.
+/obj/item/organ/liver/dwarf
+	name = "dwarf liver"
+	icon_state = "liver"
+	desc = "A dwarven liver, theres something magical about seeing one of these up close."
+	alcohol_tolerance = 0 //dwarves really shouldn't be dying to alcohol.
+	toxTolerance = 5 //Shrugs off 5 units of toxins damage.
+	maxHealth = 150 //More health than the average liver, as you aren't going to be replacing this.
+	//If it does need replaced with a standard human liver, prepare for hell.
+
+//alcohol gland
+/obj/item/organ/dwarfgland
+	name = "dwarf alcohol gland"
+	icon_state = "plasma" //Yes this is a actual icon in icons/obj/surgery.dmi
+	desc = "A genetically engineered gland which is hopefully a step forward for humanity."
+	w_class = WEIGHT_CLASS_NORMAL
+	var/stored_alcohol = 250 //They start with 250 units, that ticks down and eventaully bad effects occur
+	var/max_alcohol = 500 //Max they can attain, easier than you think to OD on alcohol.
+	var/heal_rate = 0.5 //The rate they heal damages over 400 alcohol stored. Default is 0.5 so we times 3 since 3 seconds.
+	var/alcohol_rate = 10 //Its times 0.025 making it tick down by .25 per loop per 10. EX: 20 = .50
+	var/filth_counter = 0 //Holder for the filth check cycle, basically it compares against this.
+	//These count in on_life ticks which should be 2 seconds per every increment of 1 in a perfect world.
+	var/dwarf_filth_ticker = 0 //Currently set =< 4, that means this will fire the proc around every 4-8 seconds.
+	var/dwarf_eth_ticker = 0 //Currently set =< 1, that means this will fire the proc around every 2 seconds
+
+/obj/item/organ/dwarfgland/prepare_eat()
+	var/obj/S = ..()
+	S.reagents.add_reagent("ethanol", stored_alcohol/10)
+	return S
+
+/obj/item/organ/dwarfgland/on_life() //Primary loop to hook into to start delayed loops for other loops..
+	dwarf_cycle_ticker()
+
+//Handles the delayed tick cycle by just adding on increments per each on_life() tick
+/obj/item/organ/dwarfgland/proc/dwarf_cycle_ticker()
+	if(owner.stat != DEAD) //We make sure they are not dead, so they don't increment any tickers.
+		dwarf_eth_ticker++
+		dwarf_filth_ticker++
+	if(dwarf_filth_ticker >= 4) //Should be around 4-8 seconds since a tick is around 2 seconds. 
+		dwarf_filth_cycle()		//On_life will adjust regarding other factors, so we are along for the ride.
+		dwarf_filth_ticker = 0 //We set the ticker back to 0 to go again.
+	if(dwarf_eth_ticker >= 1) //Alcohol reagent check should be around 2 seconds, since a tick is around 2 seconds.
+		dwarf_eth_cycle()
+		dwarf_eth_ticker = 0
+	//Debug messages:
+	//to_chat(owner, "<span class = 'danger'>FILTH VIEW: [filth_counter], FILTH CYCLE TICKER: [dwarf_filth_ticker]</span>")
+	//to_chat(owner, "<span class = 'danger'>ALCOHOL COUNTER: [stored_alcohol], ETH CYCLE TICKER: [dwarf_eth_ticker].</span>")
```

Comment:
remember to remove debug messages afterward.

Reply:
Removed